src/index.js

```diff
@@ -387,14 +403,209 @@ function fixTouchableMixin(t, path) {
   });
 }
 
+function isReactFragment(openingElement) {
+  if (openingElement.isJSXFragment()) {
+    return true;
+  }
+
+  if (!openingElement.node || !openingElement.node.name) return;
+
+  if (
+    openingElement.node.name.name === 'Fragment' ||
+    openingElement.node.name.name === 'React.Fragment'
+  )
+    return true;
+
+  if (
+    !openingElement.node.name.type ||
+    !openingElement.node.name.object ||
+    !openingElement.node.name.property
+  )
+    return;
+
+  return (
+    openingElement.node.name.type === 'JSXMemberExpression' &&
+    openingElement.node.name.object.name === 'React' &&
+    openingElement.node.name.property.name === 'Fragment'
+  );
+}
+
+function extendExistingRef(path) {
+  // process existing ref value
+  if (path.node.name.name === 'ref') {
+    // ignore if the ref value is already applyFSPropertiesWithRef
+    if (
+      t.isJSXExpressionContainer(path.node.value) &&
+      !(
+        t.isCallExpression(path.node.value.expression) &&
+        path.node.value.expression.callee.name === applyFSPropertiesWithRef
+      )
+    ) {
+      // only process refs on base components and applyFSPropertiesWithRef must be imported
+      if (
+        !elements.includes(path.parent.name.name) ||
+        !path.scope.hasBinding(applyFSPropertiesWithRef)
+      )
+        return;
+
+      const originalRef = path.node.value.expression;
+
+      path.replaceWith(
+        t.jsxAttribute(
+          t.jsxIdentifier('ref'),
+          t.JSXExpressionContainer(
+            t.CallExpression(t.identifier(applyFSPropertiesWithRef), [originalRef]),
+          ),
+        ),
+      );
+    }
+  }
+}
+
 /* eslint-disable complexity */
 export default function ({ types: t }) {
   return {
+    name: 'fullstory-react-native',
     visitor: {
       /* Looks like we don't currently need to add this to the interface declaration
       InterfaceDeclaration(path, state) {
         fixUIManagerJSInterface(path);
       }, */
+
+      Program: {
+        enter(path, { file, opts }) {
+          if (opts.isNewArchitectureEnabled) {
```

Comment:
I like the early exit like you did here (although I think we generally still prefer curly braces )
https://github.com/fullstorydev/fullstory-babel-plugin-react-native/pull/29/files#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R533

Reply:
To clarify, did you want me to change this piece of code to say:
```
if (!opts.isNewArchitectureEnabled) {
  return;
}
```
?

Reply:
I believe that is what Prashanth is asking for.  I call this idiom, 'handle the exception, not the rule': that is to say, code that takes 'the normal path' should be in a straight line and indented as little as possible, and 'unusual' control flow should be highlighted instead.  In this case, this code is 'expected' to run with New Architecture turned on, and so not-New Architecture is 'unusual', and hence would warrant the early return.

Reply:
Yeah, especially since there is a similar early exit down below. We should be consistent.